sdk/cosmos/azure-cosmos/azure/cosmos/_container_recreate_retry_policy.py

```diff
@@ -0,0 +1,165 @@
+            elif partition_key_definition and partition_key_definition["kind"] == "MultiHash":
+                new_partition_key = json.dumps(options["partitionKey"], separators=(',', ':'))
+            else:
+                new_partition_key = json.dumps([options["partitionKey"]])
+        return new_partition_key
+
+    async def _extract_partition_key_async(self, client: Optional[Any],
+                                           container_cache: Optional[Dict[str, Any]],
+                                           body: str) -> Optional[Union[str, List, Dict]]:
+        partition_key_definition = container_cache["partitionKey"] if container_cache else None
+        body_dict = self.__str_to_dict(body)
+        new_partition_key: Optional[Union[str, List, Dict]] = None
+        if body_dict:
+            options = await client._AddPartitionKey(self.container_link, body_dict, {}) if client else {}
+            # if partitionKey value is Undefined, serialize it as [{}] to be consistent with other SDKs.
+            if isinstance(options["partitionKey"], _Undefined):
+                new_partition_key = [{}]
+            # If partitionKey value is Empty, serialize it as [], which is the equivalent sent for migrated collections
+            elif isinstance(options["partitionKey"], _Empty):
+                new_partition_key = []
+            # else serialize using json dumps method which apart from regular values will serialize None into null
+            elif partition_key_definition["kind"] == "MultiHash":
+                new_partition_key = json.dumps(options["partitionKey"], separators=(',', ':'))
+            else:
+                new_partition_key = json.dumps([options["partitionKey"]])
+        return new_partition_key
+
+    def should_update_throughput_link(self, body: Optional[str], cached_container: Optional[Dict[str, Any]]) -> bool:
+        body_dict = self.__str_to_dict(body) if body else None
+        if not body_dict:
+            return False
+        try:
+            # If this is a request to get throughput properties then we will update the link
+            if body_dict["query"] == "SELECT * FROM root r WHERE r.resource=@link":
+                self.link = cached_container["_self"] if cached_container else None
+                return True
+        except (TypeError, IndexError, KeyError):
+            return False
+        return False
+
+    def _update_throughput_link(self, body: str) -> str:
+        body_dict = self.__str_to_dict(body) if body else None
+        if not body_dict:
+            return body
+        body_dict["parameters"][0]["value"] = self.link
+        return json.dumps(body_dict, separators=(',', ':'))
+
+    def __str_to_dict(self, dict_string: str) -> Dict:
+        try:
+            # Use ast.literal_eval() to convert string to dictionary
+            dict_obj = ast.literal_eval(dict_string)
```

Comment:
Why are we using this and not json.loads?
Can you give me an example of the string value here?